### PR TITLE
don't codegen constexpr for xnumel and rnumel

### PR DESCRIPTION
**Pull request by ngimel**
Tentative fix for #1358, that looks like triton bug. The same kernel either works or doesn't, depending on whether xnumel and rnumel are annotated as tl.constexpr. Note, this regresses perf a bit for static shapes (e.g. hf_Bert goes from 1.131 to 1.125) but correctness is more important, and we'd move to dynamic shapes soon anyway. 
Triton issue https://github.com/openai/triton/issues/714

### CODE REVIEW

File: torchinductor/codegen/triton.py [hunk 1]
```diff
@@ -934,11 +934,7 @@ def codegen_kernel(self, name=None):
 
         argdefs, _ = self.args.python_argdefs()
 
-        if config.dynamic_shapes:
-            maybe_const = ""
-        else:
-            maybe_const = ": tl.constexpr"
-
+        maybe_const = ""
```
Comment (voznesenskym) [inline torchinductor/codegen/triton.py:937]: its only used on L940, you can probably just rm the whole thing. 

Or leave it commented out with an explanation?

Comment (ngimel) [reply torchinductor/codegen/triton.py:937]: Yeah I didn't remove it because we might want to go back to it, I'll leave a comment.

Comment (voznesenskym) [reply torchinductor/codegen/triton.py:937]: We also have `argdefs.append(f"{tree.prefix.upper()}BLOCK : tl.constexpr")` - is that one alright?

Comment (ngimel) [reply torchinductor/codegen/triton.py:939]: Yes those need to be constexprs and powers of 2 - triton codegen wouldn't work otherwise.

**Comment by ngimel** [comment]
hm, test failure is really surprising - so sometimes `constexpr` is actually making things correct? (Not really correct, that model was failing accuracy check, so who knows what was happening, but at least no IMA). I'll look into this.

**Comment by jansel** [comment]
This conflicts with the new runtime, but I also included this fix into the new runtime already -- so I think we can close this.

**Pull request closed without merging**